pyglmnet/pyglmnet.py

```diff
@@ -714,20 +716,30 @@ def fit(self, X, y):
             ActiveSet = np.ones(n_features + 1)     # init active set
             z = beta[0] + np.dot(X, beta[1:])       # cache z
 
-        # Initialize loss accumulators
-        L, DL = list(), list()
+        # Iterative updates
         for t in range(0, self.max_iter):
             if self.solver == 'batch-gradient':
                 grad = _grad_L2loss(self.distr,
                                     alpha, self.Tau,
                                     reg_lambda, X, y, self.eta,
                                     beta)
-
+                if t > 1:
+                    if np.linalg.norm(grad) / np.linalg.norm(beta) < tol / lr:
+                        msg = ('\tConverged in {0:d} iterations'.format(t))
+                        logger.info(msg)
+                        break
                 beta = beta - self.learning_rate * grad
+
             elif self.solver == 'cdfast':
+                beta_old = deepcopy(beta)
                 beta, z = \
                     self._cdfast(X, y, z, ActiveSet, beta, reg_lambda)
-
+                if t > 1:
+                    if ((np.linalg.norm(beta - beta_old) /
+                         np.linalg.norm(beta_old) < tol / lr)):
```

Comment:
The denominator here is `norm(beta_old)` but in the other solver it is `norm(beta)`. Is that expected?